gapis/api/vulkan/api/queue.api

```diff
@@ -113,9 +113,15 @@ cmd VkResult vkQueueSubmit(
           wait_semaphores_all_valid.b = false
           vkErrorInvalidSemaphore(ws)
         } else {
-          LastBoundQueue.PendingCommands[len(LastBoundQueue.PendingCommands)]
-          = new!CommandReference(as!VkCommandBuffer(0), 0, cmd_vkNoCommand, 0,
-            Unsignal,    ws,    null, as!VkFence(0))
+          semaphore := Semaphores[ws]
+
+          if(semaphore.State != Submitted) {
+            vkErrorInvalidSemaphore(ws)
```

Comment:
You need a better error here

Reply:
We should create another error, vkErrorSempahoreNotSubmitted().